x-pack/plugins/security_solution/README.md

```diff
@@ -0,0 +1,130 @@
+# Security Solution
+
+Welcome to the Kibana Security Solution plugin! This README will go over getting started with development and testing.
+
+## Development
+
+## Tests
+
+The endpoint specific tests leverage the ingest manager to install the endpoint package. Before the api integration
+and functional tests are run the ingest manager is initialized. This initialization process includes reaching out to
+a package registry service to install the endpoint package. The endpoint tests support three different ways to run
+the tests given the constraint on an available package registry.
+
+1. Using Docker
+2. Running your own local package registry
+3. Using the default external package registry
+
+These scenarios will be outlined the sections below.
+
+### Endpoint API Integration Tests Location
+
+The endpoint api integration tests are located [here](../../test/security_solution_endpoint_api_int)
+
+### Endpoint Functional Tests Location
+
+The endpoint functional tests are located [here](../../test/security_solution_endpoint)
+
+### Using Docker
+
+To run the tests using the recommended docker image version you must have `docker` installed. The testing infrastructure
+will stand up a docker container using the image defined [here](../../test/ingest_manager_api_integration/config.ts#L15)
+
+Make sure you're in the Kibana root directory.
+
+#### Endpoint API Integration Tests
+
+In one terminal, run:
+
+```bash
+INGEST_MANAGEMENT_PACKAGE_REGISTRY_PORT=12345 yarn test:ftr:server --config x-pack/test/security_solution_endpoint_api_int/config.ts
```

Comment:
what port is this?

Reply:
That's just an example. It can be any port you like as long as something hasn't been bound to it already.

Reply:
oh what I mean is, what is the purpose of the port?